sway/input/text_input.c

```diff
@@ -316,6 +317,12 @@ static void input_popup_update(struct sway_input_popup *popup) {
 		parent.height = view->geometry.height;
 	}
 	popup->scene_tree = wlr_scene_subsurface_tree_create(root->layer_tree, popup->popup_surface->surface);
+	if (!scene_descriptor_assign(&popup->scene_tree->node, 
+				SWAY_SCENE_DESC_VIEW, popup)) {
```

Comment:
I expect this code to crash. This should be the `view`, not the `popup`.

Reply:
emm. what do you mean `view`, is it `root`?

Reply:
Should be:
```
scene_descriptor_assign(&popup->scene_tree->node, 
				SWAY_SCENE_DESC_VIEW, view)
```

The code expects for a `SWAY_SCENE_DESC_VIEW` to have a pointer to a `struct sway_view`

Reply:
then can I make it `SWAY_SCENE_DESC_POPUP`?

Reply:
but I make ti to view, it also coredump..

Reply:
now is that ok?

Reply:
Hmm, DESC_POPUP doesn't work as-is because it expects a `sway_xdg_popup` type. That's why I originally didn't suggest it. But DESC_POPUP has advantages for us: On transaction, popups will automatically be rearranged to follow surfaces. Plus, we could reuse the popup root layer instead of building one for `input-method`. Actually, the implementation wouldn't be correct if we don't do this. We should introduce some sort of common struct like:

```
struct sway_popup {
    struct sway_view *view;
    struct wlr_scene_node *node;
}
```

And the change `arrange_popup` in `transaction.c` to understand this new struct and port `sway_xdg_popup` to use it and input_method to use it as well. input_method popups aren't that different from xdg popups, therefore we should handle them similarly.

It should be okay to use this common layer, because input_method should always map after xdg popups in the case that a popup wants to show an input_method popup itself.

Reply:
but when it is layer_shell, how to handle it?

Reply:
ok, thanks for you advice, seems it is working well!

Reply:
Sorry, I didn't respond because I was thinking through the problem some more. I came up with this:
https://github.com/swaywm/sway/pull/7929

Reply:
Ok, I have change to desc now, can you review it now?